libmisc/user_busy.c

```diff
@@ -106,6 +107,54 @@ static int user_busy_utmp (const char *name)
 #endif				/* !__linux__ */
 
 #ifdef __linux__
+#ifdef ENABLE_SUBIDS
+#define in_parentuid_range(uid) ((uid) >= parentuid && (uid) < parentuid + range)
+static int in_mapping (const char *sname, unsigned long ruid, unsigned long euid, unsigned long suid)
+{
+	/* 41: /proc/xxxxxxxxxx/task/xxxxxxxxxx/uid_map + \0 */
```

Comment:
Not a critique of your patch as I see the existing case of this.  But I worry about the assumption that each pid is 9 characters.  One day the kernel might bump that number.  Then an old shadow might have trouble on newer kernel, and boy would that suck.

@brauner what do you think, is it worth bumping these values throughout the tree? (if they exist elsewhere)

Reply:
Actually it is kind-of hard to imagine that the PID would be larger than UINT32_MAX (when today it is just 16 bit number) - which fits into 10 digits. The comment is not really precise here because we are not looking at threads. I'll adjust it.

Reply:
Oh, disregard what I said about the comment not being precise. We actually are looking at threads. But anyway. The reserved space is 10 digits for for PID where currently it has just 5 digits.

Reply:
Yeah it seems unlikely.  But bumping the # to 100 would be so easy, and the pain if it *does* happen would be so ugly, it seems maybe worth it.

But I don't mean for that to happen in your patch, sorry for derailing this.